src/Core/Managed/Shared/Channel/TelemetryBuffer.cs

```diff
@@ -15,19 +15,21 @@ internal class TelemetryBuffer
         public Action OnFull;
 
         private const int DefaultCapacity = 500;
+        private const int DefaultMaximumUnsentBacklogSize = 1000000;
```

Comment:
I would lower the default a lot.
Even a factor of 10x the default capacity feels high and that would be still be only 1/200th of a million.